packages/cozy-client/src/models/permission.js

```diff
@@ -0,0 +1,139 @@
+import intersection from 'lodash/intersection'
+import get from 'lodash/get'
+import CozyClient from 'cozy-client'
+import { getParentFolderId } from './file'
+
+/**
+ * @typedef {object} Document - Couchdb document like an io.cozy.files
+ * @property {string} _id
+ * @property {string} _type
+ */
+
+/**
+ * @typedef {('ALL'|'GET'|'PATCH'|'POST'|'PUT'|'DELETE')} PermissionVerb
+ */
+
+/**
+ * @typedef {object} PermissionItem
+ * @property {PermissionVerb[]} verbs - ALL, GET, PUT, PATCH, DELETE, POST…
+ * @property {string} selector - defaults to `id`
+ * @property {string[]} values
+ * @property {string} type - a couch db database like 'io.cozy.files'
+ */
+
+/**
+ * Is this permission read only ?
+ *
+ * @private
+ * @param {PermissionItem} perm - permission node in a io.cozy.permissions document
+ * @param {object} options -
+ * @param {PermissionVerb[]} options.writability  -
+ * @returns {boolean} true if the note should be displayed read-only
+ */
+export function isReadOnly(perm, options = {}) {
+  const { writability = ['PATCH', 'POST', 'PUT', 'DELETE'] } = options
+  return (
+    perm.verbs && // no verbs is equivalent to ['ALL']
+    perm.verbs.length > 0 && // empty array is equivalent to ['ALL']
+    intersection(perm.verbs, ['ALL', ...writability]).length === 0
+  )
+}
+
+/**
+ * Get the list of permissions blocks
+ * @private
+ */
+async function getOwn(client) {
+  const collection = client.collection('io.cozy.permissions')
+  const data = await collection.getOwnPermissions()
+  const permissions = get(data, 'data.attributes.permissions')
+  if (!permissions) throw `Can't get self permissions`
+  return Object.values(permissions)
+}
```

Comment:
I think it is clearer when method doing asynchronous things use the verb `fetch`.